A partial commit of proposed features

Sqeaky opened the pull request:
Added variable creation on startup feature, it needs some testing before final inclusions

Added a AddDirectoryOnceIndex macro to include subdirs only once without needing to include
the suprisingly complex logic needed to do that yourself. This also needs tests.

Sqeaky committed:
A partial commit of proposed features

Added variable creation on startup feature, it needs some testing before final inclusions

Added a AddDirectoryOnceIndex macro to include subdirs only once without needing to include
the suprisingly complex logic needed to do that yourself. This also needs tests.

Sqeaky commented:
The Jenkins failure is a Git error. Looks like I will be spending some time fixing the git setup on some of the CI server this weekend.

MakoEnergy commented:
This did not build on my system.  From reviewing the code, I did notice that you were using "CACHE" in one or two places I did not expect based on our talk the other day.

MakoEnergy committed:
Stop caching "AddDirectoryOnceIndex".

Removing the caching of this variable fixed the build in my use case of configuring twice before generating.

MakoEnergy commented:
I looked at it again, and removed one occurrence of caching and that fixed the build process for me.  Although the order was a bit weird.  The test build on my system built the targets in the following order:

Mezz_Test > Mezz_StaticFoundation > Mezz_Foundation > Foundation_Tester > StaticFoundationTester > Test_Tester

It's odd, but it works.

MakoEnergy committed:
This edit wasn't necessary during my tests, however it seems odd for it to exist with the other cache removed.

MakoEnergy committed:
Correcting previous commit.
